### DISCUSSION TITLE
Altering Solution after solving step

### DISCUSSION THREAD

**Discussion by KhaledNabilSharafeldin** [idaholab/moose#18229] [Q&A General] [answered]
Hello,
I was wondering if there is a way to alter the nonlinear variable solution after the solving step. I'm doing operator splitting on reaction transport model, I have a kernel for transport part, but the reactions part which solves a simple ODE. So I was wondering if there's a way in MOOSE to alter the solution after the transport problem.

Thanks,
Khaled

**Comment by GiudGiud** [comment, marked as answer]
Hi Khaled

The canon moose way is to add all kernels and solve the problem without operator splitting.
The transport and the reactions, is this for the same variable / quantity?
If not, then a transient multiapp and transferring the quantities is a proper way to do this. 

If they are for the same quantity, then I'd still recommend using a multiapp, but you are going to have to be very careful as to what equation each app is solving and what the final numerical scheme ends up being. 

Guillaume
